### DISCUSSION TITLE
Authentication providers per tenant

### DISCUSSION THREAD

**Discussion by jdesai-care** [OrchardCMS/OrchardCore#15292] [Q&A]
I need to implement different authentication per tenant. Can anyone provide any sample or documentation?

**Comment by hishamco** [comment]
Could you please elaborate so we can help you?

**Reply by jdesai-care** [reply]
I am building a portal which will be available for multiple clients via different tenants and all of them may have a different authentication/authorization requirements and IdPs Federation. I will be using Azure AD B2C as SP and I may have to pass different IdP indicator based on the URL/Tenant. Post authentication, I may have to check claims differently for each tenants.

**Reply by hishamco** [reply]
It seems you need a lot of customization in your case or consult an Orchard Core specialist

/cc @Piedone

**Reply by Piedone** [reply]
Thanks for the shoutout! Indeed, we can also help with consulting: https://lombiq.com/.

**Comment by Piedone** [comment]
The short answer is that this is possible, and if you want to use authentication features [that OC already provides](https://docs.orchardcore.net/en/latest/docs/reference/#authentication-and-user-management) then you just need to configure these differently for each tenant (these can be configured from the admin UI of the given tenant, or [via configuration providers](https://docs.orchardcore.net/en/latest/docs/reference/core/Configuration/)).

If you need anything more involved, then you'll need to develop that. However, everything built into OC is usable with different configuration per tenant (and unless you do something deliberately against multi-tenancy, this is also true for any custom module).

**Reply by jdesai-care** [reply]
Thank you, I will try these options.
